org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/ModuleSupport.java

```diff
@@ -319,4 +335,430 @@ public static boolean isMavenJavaProject(IProject project) {
     }
     return false;
   }
+
+  /**
+   * Updates M2E container and JRE container with the jpms arguments provided to the maven-compiler-plugin. The dispatch
+   * between M2E and JRE containers is done using the targeted module. If the JRE container contains the targeted module
+   * then the jpms argument will be attached to it else it will be attached to the M2E container
+   * 
+   * @param request
+   * @param classpath
+   * @param monitor
+   * @param compilerArgs
+   */
+  public static void configureRawClasspath(ProjectConfigurationRequest request, IClasspathDescriptor classpath,
+      IProgressMonitor monitor, List<String> compilerArgs) {
+
+    IMavenProjectFacade facade = request.getMavenProjectFacade();
+
+    IJavaProject javaProject = JavaCore.create(facade.getProject());
+    if(javaProject == null || !javaProject.exists() || classpath == null) {
+      return;
+    }
+
+    JpmsArgs jpmsArgs = JpmsArgs.computeFromArgs(compilerArgs);
```

Comment:
Probably return an `Optional<JpmsArgs>`, and return early if it's empty

Reply:
ok, done